main/webapp/modules/core/scripts/views/data-table/data-table-view.js

```diff
@@ -863,6 +874,32 @@ DataTableView.prototype._createSortingMenu = function(elmt) {
   MenuSystem.createAndShowStandardMenu(items, elmt, { horizontal: false });
 };
 
+var doAllFillDown = function() {
+  for (var i = 0; i < theProject.columnModel.columns.length; i++) {
+    Refine.postCoreProcess(
+        "fill-down",
+        {
+          columnName: theProject.columnModel.columns[i].name
+        },
+        null,
+        { modelsChanged: true }
+    );
+  }
+};
```

Comment:
I think it is fine to use the existing operations to implement this (so, resulting in multiple steps in the history).
However, doing a for loop like this will fire up HTTP queries to the backend in no particular order: the first column could well end up being filled down after the first one! That is a problem since the final state of the table can depend on the order of execution of these operations, if we are in records mode (simply because filling down the first column will change records, and filling down on another column in records mode respects the records boundaries).

So, the code should be adapted to use callbacks instead: the second column should only be filled down once the first one has. The `postCoreProcess` method accepts callbacks which can be used for that. Unfortunately that is going to lead to less readable code but I don't see how to avoid that to ensure correctness.

Reply:
Thanks for the advise! I have tried to chain the operations with callbacks. 

However, I am not sure if my way is the simplest/cleanest way to do it. May I get your review on this?

Reply:
I would say this is a rather elegant way to do it!